docs/porting/security/psa_targets_in_mbed-os.md

```diff
@@ -0,0 +1,561 @@
+            "MBEDTLS_PSA_HAS_ITS_IO",
+            "MBEDTLS_USE_PSA_CRYPTO"
+        ],
+        "detect_code": [
+            "190A"
+        ],
+        "post_binary_hook": {
+            "function": "PSOC6Code.sign_image"
+        },
+        "forced_reset_timeout": 5,
+        "overrides": {
+            "network-default-interface-type": "WIFI"
+        },
+        "program_cycle_s": 10,
+        "tfm_target_name": "psoc64",
+        "tfm_bootloader_supported": false,
+        "tfm_default_toolchain": "GNUARM",
+        "tfm_supported_toolchains": [
+            "GNUARM"
+        ],
+        "tfm_delivery_dir": "TARGET_Cypress/TARGET_PSOC6/TARGET_CY8CKIT_064S2_4343W",
+        "TFM_OUTPUT_EXT": "hex"
+    }
+```
+
+Please note the config options `macros_add`, `extra_labels_add` and `device_has_remove`. To add or remove `macros`, `extra_labels` or target capabilities, a PSA target definition must use
+[macros/extra_labels/device_has]`_add` or [macros/extra_labels/device_has]`_remove` (not `macros`, `extra_labels` or `device_has`).
+
+To add or remove a feature, use `feature_`[add/remove].
+
+By default, a TF-M build generates a `bin` file. If the target requires a `hex` file, you need to add the attribute `"TFM_OUTPUT_EXT": "hex"` to the target definition. The build script will convert `bin` to `hex`, then copy the `hex` to `tfm_delivery_dir`.
+
+This dual-core PSA target doesn't inherit from `PSA_DUAL_CORE` because it has to inherit from one of its family targets<!--I'm still not sure why I have to inherit from a family target-->. Hence, we have added additional attributes:
+
+```json
+        "features_add": [
+            "PSA"
+        ],
+        "extra_labels_add": [
+            "TFM",
+            "TFM_DUALCPU"
+        ],
+        "macros_add": [
+            "MBEDTLS_PSA_HAS_ITS_IO",
+            "MBEDTLS_USE_PSA_CRYPTO"
+        ]
+```
+
+If a dual-core PSA target can inherit from `PSA_DUAL_CORE`, then there is no need to add these additional attributes.
+
+### Adding Armv8-M PSA targets
+
+An Mbed OS (NSPE) target must contain the following attributes (in addition to other target attributes defined in [porting a custom board](../porting/porting-a-custom-board.html)
+and [porting targets](../porting/porting-targets.html)):
+
+* `inherits`: PSA generic target `PSA_V8_M`, unless the target has to inherit from
+  one of its family targets.
+* `tfm_target_name`: Target name in TF-M.
+* `tfm_bootloader_supported`: If TF-M bootloader is supported by the target.
+
+    The supported values are `true` and `false`.
+* `tfm_supported_toolchains`: Supported TF-M toolchains.
+
+    The supported values are `ARMCLANG` and `GNUARM`.
+* `tfm_default_toolchain`: Default TF-M toolchain.
+
+     The supported values are `ARMCLANG` and `GNUARM`.
+
+* `tfm_delivery_dir`: The directory to which TF-M binary will be copied.
+* `TFM_OUTPUT_EXT`: Optional attribute that indicates the output extension of the TF-M secure binary.
```

Comment:
```suggestion
* `tfm_output_ext`: Optional attribute that indicates the output extension of the TF-M secure binary.
```

Reply:
Unfortunately, we can't change it here. The code contains capitalised text.